projects/storefrontlib/shared/directives/horizontal-scrolling-position/horizontal-scrolling-position.directive.ts

```diff
@@ -62,8 +63,13 @@ export class HorizontalScrollingPositionDirective
   protected windowResizeSubscription: Subscription | undefined;
   protected _scrollingAreaWidth$ = new BehaviorSubject<number>(0);
 
-  public readonly scrollingAreaWidth$ =
-    this._scrollingAreaWidth$.asObservable();
+  /**
+   * Width of the scrolling area, debounced every 300ms in case of continuous resizing.
+   */
+  public readonly scrollingAreaWidth$ = this._scrollingAreaWidth$.pipe(
+    debounceTime(300),
```

Comment:
`query:` Is there any reason that it has to be 300ms exactly?

Reply:
No, it's an arbitrary value. 
I've followed the same value that is used in `WindowRef.resize$`:
https://github.com/SAP/spartacus/blob/29c6a5525ff90a37d48dbc884eab444796523f42/projects/core/src/window/window-ref.ts#L97-L102